src/GraftUKFAbsolute.cpp

```diff
@@ -404,6 +407,32 @@ double GraftUKFAbsolute::predictAndUpdate(){
 
 	graft_covariance_ = predicted_covariance - K*predicted_measurement_uncertainty*K.transpose();
 
+	for( int i=0; i<SIZE; i++ ) {
+		for( int j=0; j<SIZE; j++ ) {
+			if( !std::isfinite(graft_covariance_(i, j)) ) {
+				diverged_ = true;
+			}
+		}
+	}
+	if( diverged_ ) {
+		// print offending messages
+		std::stringstream errmsg;
+		errmsg << "Covariance diverged! Offending topics are: ";
+
+		// For each topic
+		for(size_t i = 0; i < topics_.size(); i++){
+			// Get the measurement msg and covariance
+			graft::GraftSensorResidual::ConstPtr meas = topics_[i]->z();
+			if( meas ) {
+				if( i>0 ) errmsg << ", ";
+				errmsg << topics_[i]->getName() << "(";
+				errmsg << *meas << ")";
+			}
+		}
+
+		ROS_ERROR_STREAM(errmsg.str());
```

Comment:
I would suggest throttle over once-only so that if you remotely open rqt_console or similar after the filter has diverged, you will still see the message.